src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts

```diff
@@ -110,66 +116,14 @@ export class SearchFacetFilterComponent implements OnInit, OnDestroy {
     this.currentPage = this.getCurrentPage().pipe(distinctUntilChanged());
 
     this.searchOptions$ = this.searchConfigService.searchOptions;
-    this.subs.push(this.searchOptions$.subscribe(() => this.updateFilterValueList()));
-    const facetValues$ = observableCombineLatest(this.searchOptions$, this.currentPage).pipe(
-      map(([options, page]) => {
-        return { options, page };
-      }),
-      switchMap(({ options, page }) => {
-        return this.searchService.getFacetValuesFor(this.filterConfig, page, options)
-          .pipe(
-            getFirstSucceededRemoteData(),
-            map((results) => {
-                return {
-                  values: observableOf(results),
-                  page: page
-                };
-              }
-            )
-          );
-      })
+    this.subs.push(
+      this.searchOptions$.subscribe(() => this.updateFilterValueList()),
+      this.refreshFilters.asObservable().pipe(
```

Comment:
If we can't find a better solution than the delay, I think we should at least move it to this pipe instead of the `ListableObjectComponentLoaderComponent`. Otherwise we'd be affecting a lot of other components.

I checked this locally real quick and it seems to have the same effect if we move the `debounceTime` here and leave out the `detectChanges` call.